pyramid/url.py

```diff
@@ -96,19 +96,17 @@ def _partial_application_url(self, scheme=None, host=None, port=None):
             if scheme == 'http':
                 if port is None:
                     port = '80'
-        url = scheme + '://'
-        if port is not None:
-            port = str(port)
         if host is None:
             host = e.get('HTTP_HOST')
-        if host is None:
-            host = e['SERVER_NAME']
+            if host is None:
+                host = e['SERVER_NAME']
```

Comment:
This is OK with me. With the nesting, I read this implicitly as "if host is *still* None, then...".